.pipeline/templates/api/critterbase.yaml

```diff
@@ -0,0 +1,593 @@
+          name: ${APP_NAME}-db
+        ipFamilyPolicy: SingleStack
+        sessionAffinity: None
+        type: ClusterIP
+
+    # API Route <NAMESPACE_ENVIRONMENT>
+    - kind: Route
+      apiVersion: route.openshift.io/v1
+      metadata:
+        name: ${APP_NAME}-api
+        namespace: ${NAMESPACE_PREFIX}-${NAMESPACE_ENVIRONMENT}
+        annotations:
+        labels:
+          app: ${APP_NAME}
+          name: ${APP_NAME}-api
+          tag: ${TEMPLATE_TAG}
+      spec:
+        host: moe-critterbase-api-${NAMESPACE_ENVIRONMENT}.apps.silver.devops.gov.bc.ca
+        path: /api/
+        to:
+          kind: Service
+          name: ${APP_NAME}-api
+          weight: 100
+        port:
+          targetPort: ${API_PORT}-tcp
+        tls:
+          insecureEdgeTerminationPolicy: Redirect
+          termination: edge
+        wildcardPolicy: None
+      status:
+        ingress: null
+
+    # API Deployment <NAMESPACE_ENVIRONMENT>
+    - kind: Deployment
+      apiVersion: apps/v1
+      metadata:
+        name: ${APP_NAME}-api
+        namespace: ${NAMESPACE_PREFIX}-${NAMESPACE_ENVIRONMENT}
+        annotations:
+          openshift.io/generated-by: OpenShiftWebConsole
+        generation: 0
+        labels:
+          app: ${APP_NAME}
+          name: ${APP_NAME}-api
+          tag: ${TEMPLATE_TAG}
+      spec:
+        replicas: ${{API_REPLICAS}}
+        revisionHistoryLimit: 5
+        selector:
+          matchLabels:
+            name: ${APP_NAME}-api
+        strategy:
+          type: RollingUpdate
+          rollingUpdate:
+            maxSurge: 25%
+            maxUnavailable: 25%
+        template:
+          metadata:
+            annotations: null
+            labels:
+              app: ${APP_NAME}
+              name: ${APP_NAME}-api
+              tag: ${TEMPLATE_TAG}
+          spec:
+            containers:
+              - resources:
+                  limits:
+                    cpu: ${API_CPU_LIMIT}
+                    memory: ${API_MEMORY_LIMIT}
+                  requests:
+                    cpu: ${API_CPU_REQUEST}
+                    memory: ${API_MEMORY_REQUEST}
+                readinessProbe:
+                  httpGet:
+                    path: /api
+                    port: ${{API_PORT}}
+                    scheme: HTTP
+                  # API needs to be ready in 120 seconds
+                  # Delay 30s + (3 failures X 30s) = 120 seconds
+                  initialDelaySeconds: 30
+                  periodSeconds: 30
+                  timeoutSeconds: 1
+                  successThreshold: 1
+                  failureThreshold: 3
+                # Check API alive every 60 seconds
+                livenessProbe:
+                  httpGet:
+                    path: /api
+                    port: ${{API_PORT}}
+                    scheme: HTTP
+                  initialDelaySeconds: 30
+                  periodSeconds: 30
+                  timeoutSeconds: 1
+                  failureThreshold: 1
```

Comment:
If the failureThreshold here is 1, then the note above about the pod needing to be ready in 120 seconds wont be true, as this probe will still fail in 30 seconds.

Reply:
Also, would it be 120 seconds or 90 seconds?

Initial 30 second delay, and then it depends if the first period is right after the initial delay or if it waits for 30 more seconds before doing the first call.

Reply:
It truly doesn't really matter, mostly just curious which one of these it is. I would assume the first one, but I'm not sure:

- 30 + (fail) + 30 + (fail) + 30 (fail) = 90
- 30 + 30 (fail) + 30 (fail) + 30 (fail) = 120